### ISSUE TITLE
CI: org.elasticsearch.index.shard.IndexShardIT#testIndexCanChangeCustomDataPath

### ISSUE THREAD

**Issue by jkakavas** [elastic/elasticsearch#43964]
#### Example reproduction

```

./gradlew :server:integTest --tests "org.elasticsearch.index.shard.IndexShardIT.testIndexCanChangeCustomDataPath" \
  -Dtests.seed=1CAAEC094ABB61B8 \
  -Dtests.security.manager=true \
  -Dtests.locale=he-IL \
  -Dtests.timezone=CAT \
  -Dcompiler.java=12 \
  -Druntime.java=8
```

#### Example failure

https://scans.gradle.com/s/zngdfh37brdo2/

#### Frequency

[11 times since yesterday](https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-30d,mode:quick,to:now))&_a=(columns:!(_source),index:e58bf320-7efd-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:'!'testIndexCanChangeCustomDataPath!''),sort:!(time,desc))) 

Will mute this

**Comment by ywelsch** [comment]
I've assigned this to @tlrx as I suspect it might be related to the translog trimming work which now runs on closed indices. The test assumes that the data on disk does not change for a closed  index.

**Comment by mayya-sharipova** [comment]
Another more recent failure: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+multijob-unix-compatibility/os=ubuntu-18.04&&immutable/119/console

Doesn't reproduce for me as well.

**Comment by tlrx** [comment]
Fixed in #43978
